package com.github.mikephil.charting.data.filter;

/* JADX INFO: loaded from: classes.dex */
public class Approximator {
    public float[] reduceWithDouglasPeucker(float[] fArr, float f) {
        com.github.mikephil.charting.data.filter.Approximator.Line line = new com.github.mikephil.charting.data.filter.Approximator.Line(fArr[0], fArr[1], fArr[fArr.length - 2], fArr[fArr.length - 1]);
        float f2 = 0.0f;
        int i = 0;
        for (int i2 = 2; i2 < fArr.length - 2; i2 += 2) {
            float fDistance = line.distance(fArr[i2], fArr[i2 + 1]);
            if (fDistance > f2) {
                i = i2;
                f2 = fDistance;
            }
        }
        if (f2 > f) {
            float[] fArrReduceWithDouglasPeucker = reduceWithDouglasPeucker(java.util.Arrays.copyOfRange(fArr, 0, i + 2), f);
            float[] fArrReduceWithDouglasPeucker2 = reduceWithDouglasPeucker(java.util.Arrays.copyOfRange(fArr, i, fArr.length), f);
            return concat(fArrReduceWithDouglasPeucker, java.util.Arrays.copyOfRange(fArrReduceWithDouglasPeucker2, 2, fArrReduceWithDouglasPeucker2.length));
        }
        return line.getPoints();
    }

    float[] concat(float[]... fArr) {
        int length = 0;
        for (float[] fArr2 : fArr) {
            length += fArr2.length;
        }
        float[] fArr3 = new float[length];
        int i = 0;
        for (float[] fArr4 : fArr) {
            for (float f : fArr4) {
                fArr3[i] = f;
                i++;
            }
        }
        return fArr3;
    }

    private class Line {
        private float dx;
        private float dy;
        private float exsy;
        private float length;
        private float[] points;
        private float sxey;

        public Line(float f, float f2, float f3, float f4) {
            float f5 = f - f3;
            this.dx = f5;
            float f6 = f2 - f4;
            this.dy = f6;
            this.sxey = f * f4;
            this.exsy = f3 * f2;
            this.length = (float) java.lang.Math.sqrt((f5 * f5) + (f6 * f6));
            this.points = new float[]{f, f2, f3, f4};
        }

        public float distance(float f, float f2) {
            return java.lang.Math.abs((((this.dy * f) - (this.dx * f2)) + this.sxey) - this.exsy) / this.length;
        }

        public float[] getPoints() {
            return this.points;
        }
    }
}
